src/serac/physics/base_physics.hpp

```diff
@@ -130,6 +130,16 @@ class BasePhysics {
    */
   virtual void outputState() const;
 
+  /**
+   * @brief Get an mfem::Operator that calculates the residual at the current outputState
+   */
+  virtual mfem::Vector outputResidual() = 0;
```

Comment:
What would this be used for? Solving adjoint equations or something else? It may be easier to swap implementations if we just have something like `solveAdjoint` or `solveEigensystem` instead of giving direct access to the underlying objects.

Reply:
I was mainly using it to test the `outputGradient` method, but I agree it might only be used in checking that a solution is "converged" enough. 

I think it's good to have something that is like `outputGradient()` if that's what you actually want (e.g. you want to specifically grab a tangent matrix). It's also good for checking when adding new linear algebra solvers. 

I think something like `solveAdjoint` would be useful for the adjoint case as you could specifically better handle how dof's are ordered and so forth.